doc/conf.py

```diff
@@ -891,6 +894,68 @@ def setup(app):
     }
 
 
+def infer_next_release_versions():
+    """Infer the most likely next release versions to make."""
+    all_version_full = {"rc": "0.99.0rc1", "final": "0.99.0", "bf": "0.98.1"}
+    all_version_main = {"rc": "0.99", "final": "0.99", "bf": "0.98"}
+    all_previous_tag = {"rc": "X.Y.Z", "final": "X.Y.Z", "bf": "X.Y.Z"}
+
+    try:
+        # Fetch the version switcher JSON; see `html_theme_options` for more details
+        versions_json = json.loads(
+            urlopen(html_theme_options["switcher"]["json_url"], timeout=10).read()
```

Comment:
I think we can simplify the logic a lot. We never have rc versions in the docs. It only contains
- the dev version
- the current stable version
- the last stable version of previous majors/minors, i.e there will be 1.4.2 but no 1.4.1 nor 1.4.0.

So I think we want to
- for RC: find the one with `dev` in its name (it's the first one) and replace `dev0` by `0rc1` in the version number.
- for Major/Minor: find the one with `dev` in its name (it's the second one) and replace `dev0` by `0` in the version number.
- for bug fix: find the one with `stable` in its name and increase the last version digit by 1.
- if we need to reference the last stable version, we take the third one.

Reply:
Thanks for the suggestion! Updated in https://github.com/scikit-learn/scikit-learn/pull/29412/commits/94d98a79a31c4d2d9f317719fe6d105ec48b7749